gender_analysis/analysis/frequency.py

```diff
@@ -0,0 +1,463 @@
+    return output
+
+
+def _get_gender_word_frequencies_relative(gender_word_counts: GenderCounts) -> GenderFrequencies:
+    """
+    A private helper function that examines identifier counts keyed to Gender instances,
+    determines the total count value of all identifiers across Gender instances,
+    and returns the percentage of each identifier count over the total count.
+
+    :param gender_word_counts: a dictionary keying gender instances to string identifiers keyed to
+                               integer counts.
+    :return: a dictionary with the integer counts transformed into float values representing
+             the identifier count as a percentage of the total identifier counts across all
+             identifiers.
+
+    >>> from gender_analysis.gender.common import MALE, FEMALE
+    >>> test_input = {MALE: {'he': 5, 'him': 2}, FEMALE: {'she': 3, 'her': 10}}
+    >>> _get_gender_word_frequencies_relative(test_input)
+    {<Male>: {'he': 0.25, 'him': 0.1}, <Female>: {'she': 0.15, 'her': 0.5}}
+    """
+
+    output = {}
+    total_word_count = 0
+    for gender in gender_word_counts:
+        for word in gender_word_counts[gender]:
+            total_word_count += gender_word_counts[gender][word]
+
+    for gender in gender_word_counts:
+        output[gender] = {}
+        for word, original_count in gender_word_counts[gender].items():
+            try:
+                frequency = original_count / total_word_count
+            except ZeroDivisionError:
+                frequency = 0
+            output[gender][word] = frequency
+
+    return output
+
+
+class GenderFrequencyAnalyzer(CorpusAnalyzer):
+    """
+    The GenderFrequencyAnalyzer instance accepts a series of texts and a series of Gender instances
+    and finds occurrences of each of the Gender instances' identifiers (currently pronouns).
+    Helper methods are provided to organize and analyze those occurrences according to
+    relevant criteria.
+
+    Instance methods:
+        by_date()
+        by_document()
+        by_gender()
+        by_identifier()
+        by_metadata()
+    """
+
+    def __init__(self, genders: Optional[Sequence[Gender]] = None, **kwargs) -> None:
+        """
+        Initializes a GenderFrequencyAnalyzer object that can be used for retrieving
+        analyses concerning the number of occurrences of gendered pronouns.
+
+        :param texts: a Corpus, Document, or list of Documents.
+        :param genders: a list of Gender instances.
+        """
+
+        super().__init__(**kwargs)
+
+        if genders is None:
+            genders = BINARY_GROUP
+
+        if not all(isinstance(item, Gender) for item in genders):
+            raise ValueError('all items in list genders must be of type Gender')
+
+        self.genders = genders
+
+        count = {}
+        frequencies = {}
+        relatives = {}
+
+        for document in self.corpus:
```

Comment:
For later: In #168, I moved the actual analysis code for two modules out of the init into a separate `run_analysis()` method. We might get in this habit, to make any later rerunning of the analysis (e.g., if the corpus changes) easier.